src/api/DatabaseAccess/CourseDbContext.cs

```diff
@@ -247,19 +247,21 @@ public List<FeeCaps> GetFeeCaps()
 
         public List<Provider> SuggestProviders(string query)
         {
-            return Providers.FromSql(@"
+            var providers =  Providers.FromSql(@"
 SELECT * FROM (
     SELECT ""provider"".*, COUNT(*) AS cnt
     FROM ""provider""
     JOIN ""course"" ON ""course"".""ProviderId"" = ""provider"".""Id""  OR ""course"".""AccreditingProviderId"" = ""provider"".""Id""
     WHERE (to_tsvector('english', ""provider"".""Name"") @@ to_tsquery('english', quote_literal(@query) || ':*')) IS TRUE
     OR (to_tsvector('english', ""provider"".""ProviderCode"") @@ to_tsquery('english', quote_literal(@query) || ':*')) IS TRUE
     GROUP BY ""provider"".""Id"") AS sub
-ORDER BY lower(""Name"") <> lower(@query) ASC, ""cnt"" DESC
-LIMIT @limit",
-            new NpgsqlParameter("@query", query.Replace(@"\","")),
-            new NpgsqlParameter("@limit", 5))
+ORDER BY lower(""Name"") <> lower(@query) ASC, ""cnt"" DESC",
```

Comment:
can remove the order by `lower(""Name"") <> lower(@query)` clause because it's bringing exact matches to top, which is basically a simpler version of what you are doing below.

Remove because complexity--

Reply:
keep limit but set to 10 to minimize db/app traffic